Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocRowRendererTest.cs

```diff
@@ -435,6 +451,65 @@ public void TestDiagnosticMetadataRendering ()
       Html.AssertAttribute(tr, DiagnosticMetadataAttributesForObjectBinding.BocListRowIndex, 1.ToString());
     }
 
+    [Test]
+    public void RenderValidationRow_ShowsError ()
+    {
+      var validationSummaryRenderer = new BocListValidationSummaryRenderer(RenderingFeatures.Default, _bocListCssClassDefinition);
+
+      var validationFailureRepository = new BocListValidationFailureRepository();
+      validationFailureRepository.AddValidationFailuresForDataRow(BusinessObject, new []{BusinessObjectValidationFailure.Create("Big error")});
+
+      List.Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepository);
+
+      IBocRowRenderer renderer = new BocRowRenderer(
+          _bocListCssClassDefinition,
+          new BocIndexColumnRenderer(RenderingFeatures.Default, _bocListCssClassDefinition),
+          new BocSelectorColumnRenderer(RenderingFeatures.Default, _bocListCssClassDefinition, _stubLabelReferenceRenderer),
+          RenderingFeatures.Default,
+          validationSummaryRenderer);
+      var columnRenderers = new[]
+                            {
+                                new BocColumnRenderer(
+                                    new StubColumnRenderer(new FakeResourceUrlFactory()),
+                                    new StubColumnDefinition(),
+                                    columnIndex: 0,
+                                    visibleColumnIndex: 5,
+                                    isRowHeader: true,
+                                    showIcon: false,
+                                    SortingDirection.None,
+                                    orderIndex: -1)
+                            };
+      var businessObjectWebServiceContext = BusinessObjectWebServiceContext.Create(null, null, null);
+      Html.Writer.RenderBeginTag("root");
+      renderer.RenderDataRow(
+          new BocListRenderingContext(HttpContext, Html.Writer, List.Object, businessObjectWebServiceContext, columnRenderers),
+          new BocListRowRenderingContext(new BocListRow(0, BusinessObject), 1, false),
+          17);
+
+      Html.Writer.RenderEndTag();
+
+      var document = Html.GetResultDocument();
+
+      var root = Html.GetAssertedChildElement(document, "root", 0);
+
+      var trRow = Html.GetAssertedChildElement(root, "tr", 0);
+      var trValidationRow = Html.GetAssertedChildElement(root, "tr", 1);
+
+      var td = Html.GetAssertedChildElement(trValidationRow, "td", 0);
+      td.AssertAttributeValueEquals("colspan", "6");
+
+      var outerDiv = Html.GetAssertedChildElement(td, "div", 0);
+
+      var innerDiv = Html.GetAssertedChildElement(outerDiv, "div", 0);
+
+      var ul = Html.GetAssertedChildElement(innerDiv, "ul", 0);
+      var li = Html.GetAssertedChildElement(ul, "li", 0);
+
+      var span = Html.GetAssertedChildElement(li, "span", 0);
+
+      Assert.That(span.InnerText, Is.EqualTo("Big error"));
```

Comment:
Please move to separate test for BocListValidationSummaryRenderer. There shold be more test code anway. It's possible that we need to move these tests into a separate task so we can finish up here.

We can use a stub-implementation here, that just renders a dummy tag with relevant information. It's also okay (we just didn't do this yet) that we capture the BocListValidationSummaryRenderArguments in the stub-object and then assert the BocListValidationSummaryRenderArguments after rendering has completed. This would depend on how many details we want to assert.

Reply:
There are already tests for the SummaryRenderer, if there were too little, we should probably add them from the other ticket/ created a ticket during the other ticket.

Reply:
ah, ok. didn't check this PR in detail. maybe quick look-see tomorrow (Wednesday)